src/Sylius/Behat/Context/Ui/Admin/ManagingZoneContext.php

```diff
@@ -0,0 +1,422 @@
+use Sylius\Behat\Service\NotificationCheckerInterface;
+use Sylius\Behat\NotificationType;
+use Sylius\Component\Addressing\Model\ZoneInterface;
+use Sylius\Component\Addressing\Model\ZoneMemberInterface;
+use Webmozart\Assert\Assert;
+
+/**
+ * @author Arkadiusz Krakowiak <arkadiusz.krakowiak@lakion.com>
+ */
+final class ManagingZoneContext implements Context
+{
+    const RESOURCE_NAME = 'zone';
+
+    /**
+     * @var CreatePageInterface
+     */
+    private $createPage;
+
+    /**
+     * @var IndexPageInterface
+     */
+    private $indexPage;
+
+    /**
+     * @var UpdatePageInterface
+     */
+    private $updatePage;
+
+    /**
+     * @var CurrentPageResolverInterface
+     */
+    private $currentPageResolver;
+
+    /**
+     * @var NotificationCheckerInterface
+     */
+    private $notificationChecker;
+
+    /**
+     * @param CreatePageInterface $createPage
+     * @param IndexPageInterface $indexPage
+     * @param UpdatePageInterface $updatePage
+     * @param CurrentPageResolverInterface $currentPageResolver
+     * @param NotificationCheckerInterface $notificationChecker
+     */
+    public function __construct(
+        CreatePageInterface $createPage,
+        IndexPageInterface $indexPage,
+        UpdatePageInterface $updatePage,
+        CurrentPageResolverInterface $currentPageResolver,
+        NotificationCheckerInterface $notificationChecker
+    ) {
+        $this->createPage = $createPage;
+        $this->indexPage = $indexPage;
+        $this->updatePage = $updatePage;
+        $this->currentPageResolver = $currentPageResolver;
+        $this->notificationChecker = $notificationChecker;
+    }
+
+    /**
+     * @Given I want to create a new zone consisting of :memberType
+     */
+    public function iWantToCreateANewZoneWithMembers($memberType)
+    {
+        $this->createPage->open(['type' => $memberType]);
+    }
+
+    /**
+     * @Given I want to see all zones in store
+     */
+    public function iWantToSeeAllZonesInStore()
+    {
+        $this->indexPage->open();
+    }
+
+    /**
+     * @Given /^I want to modify the (zone named "[^"]+")$/
+     */
+    public function iWantToModifyAZoneNamed(ZoneInterface $zone)
+    {
+        $this->updatePage->open(['id' => $zone->getId()]);
+    }
+
+    /**
+     * @When /^I delete (zone named "([^"]*)")$/
+     */
+    public function iDeleteZoneNamed(ZoneInterface $zone)
+    {
+        $this->indexPage->open();
+        $this->indexPage->deleteResourceOnPage(['name' => $zone->getName(), 'code' => $zone->getCode()]);
+    }
+
+    /**
+     * @When /^I remove (the "([^"]*)" province member)$/
+     * @When /^I remove (the "([^"]*)" country member)$/
+     * @When /^I remove (the "([^"]*)" zone member)$/
+     */
+    public function iRemoveTheMember(ZoneMemberInterface $zoneMember)
+    {
+        $this->updatePage->removeMember($zoneMember);
+    }
+
+    /**
+     * @When I rename it to :name
+     */
+    public function iRenameItTo($name)
```

Comment:
For this one and method below, what do you think about merging them and using current page resolver? This solution is also good, but just a though